public/apps/login/login-page.tsx

```diff
@@ -220,9 +220,19 @@ export function LoginPage(props: LoginPageDeps) {
         }
         case AuthType.SAML: {
           const samlConfig = props.config.ui[AuthType.SAML].login;
-          formBodyOp.push(
-            renderLoginButton(AuthType.SAML, SAML_AUTH_LOGIN_WITH_FRAGMENT, samlConfig)
-          );
+          const urlParams = new URLSearchParams(window.location.search);
+          let nextUrl = urlParams.get('nextUrl');
```

Comment:
Confusing to see this variable assigned values multiple times.

Reply:
The function should be simpler now. I reduced the amount of usages of nextUrl